sdk/storage_datalake/src/clients/data_lake_client.rs

```diff
@@ -1,31 +1,28 @@
 use crate::clients::FileSystemClient;
-use crate::requests::*;
+use crate::operations::ListFileSystems;
 use crate::shared_key_authorization_policy::SharedKeyAuthorizationPolicy;
-use azure_core::{ClientOptions, HttpClient, Pipeline};
-use azure_storage::core::prelude::*;
-use azure_storage::storage_shared_key_credential::StorageSharedKeyCredential;
-use bytes::Bytes;
-use http::method::Method;
-use http::request::{Builder, Request};
+use azure_core::{ClientOptions, Context, HttpClient, Pipeline};
+use azure_storage::core::storage_shared_key_credential::StorageSharedKeyCredential;
+use azure_storage::core::{clients::ServiceType, Result};
+use http::request::Builder;
 use std::sync::Arc;
 
 const DEFAULT_DNS_SUFFIX: &str = "dfs.core.windows.net";
 
 #[derive(Debug, Clone)]
 pub struct DataLakeClient {
     pipeline: Pipeline,
-    storage_client: Arc<StorageClient>,
     custom_dns_suffix: Option<String>,
     url: String, // TODO: Use CloudLocation similar to CosmosClient
+    pub(crate) context: Context,
 }
 
 impl DataLakeClient {
     pub(crate) fn new_with_options(
-        storage_client: Arc<StorageClient>,
         credential: StorageSharedKeyCredential,
         custom_dns_suffix: Option<String>,
         options: ClientOptions,
-    ) -> Self {
+    ) -> Result<Self> {
```

Comment:
I don't see why this made failable. You mention url parsing in the PR description. Is that happening here? I don't see it.

Reply:
Not sure what happened here, this does of course not need to be fallible. WE have a panic in the constructor for `FileSystemClient` where we actually do parse the url. Removed this since the panic will be gone in the next PR.